### PR TITLE
Add compiler option c++17 & link path to python for macOS

### PR DESCRIPTION
**Pull request by iSwapna**
The kernel compile code needed the c++17 option and the python library to be linked to work on MacOS. Making the PR a draft to see whether CI runs into issues before submitting for review.

### CODE REVIEW

**Comment by iSwapna** [comment]
@sh1r1sh

**Comment by parsifal-47** [comment]
Triton-shared depends on triton and triton does not support macOS, it only supports Linux

**Comment by iSwapna** [comment]
> Triton-shared depends on triton and triton does not support macOS, it only supports Linux. If macOS is supported, we need to setup a test

They do seem to run CI for macOS: https://github.com/triton-lang/triton/blob/759275dc9b5620c7797e16f1185a4911d5cbd7de/.github/workflows/integration-tests.yml#L35

Should I add a test setup? Not sure of the protocol here. Please advise. Thanks for the quick review!

**Comment by parsifal-47** [comment]
> They do seem to run CI for macOS: https://github.com/triton-lang/triton/blob/759275dc9b5620c7797e16f1185a4911d5cbd7de/.github/workflows/integration-tests.yml#L35
> 
> Should I add a test setup? Not sure of the protocol here. Please advise. Thanks for the quick review!

Interesting! and a bit confusing because it says "runs-on: ubuntu-latest", anyway, if it works for you let's try adding it here:
https://github.com/microsoft/triton-shared/tree/main/.github/workflows
Unfortunately, I do not have permissions to trigger it or approve, you'll need to wait for a maintainer's review. Thank you!

**Comment by nhat-nguyen** [comment]
@iSwapna Thanks for the contributions and @parsifal-47 for the quick review. Changes look good to me. Another macOS build on the pipeline would be ideal but not required for now.

**Review by nhat-nguyen — APPROVED** [review]

**Pull request merged**
